File: apps/fsharp/src/MachineLearning/Chapter11/Main.fs
module MachineLearning.Chapter11.Main

open System.IO
open MachineLearning.Dataset
open MachineLearning.Chapter11.Experiments

/// Survived（決定木）と cinema（線形回帰）を、K 分割交差検証の平均で評価して表示する
let run (print: string -> unit) : unit =
    print $"Survived（決定木、{NSplits} 分割交差検証の平均）"

    for name, score in evaluateSurvived (Path.Combine(dataDir (), "Survived.csv")) do
        print $"  {name}: {score:F4}"

    print $"cinema（線形回帰、{NSplits} 分割交差検証の平均）"

    for name, score in evaluateCinema (Path.Combine(dataDir (), "cinema.csv")) do
        print $"  {name}: {score:F2}"

File: apps/scala/src/main/scala/machinelearning/chapter11/Main.scala
package machinelearning.chapter11

import java.nio.file.Paths
import java.util.Locale
import machinelearning.dataset.DataDir

/** Survived.csv と cinema.csv を K 分割交差検証で評価し、指標ごとの平均を表示する。 */
object Main:

  def run(print: String => Unit): Unit =
    val dataDir = DataDir.current()
    print(s"Survived（決定木、${Experiments.NSplits} 分割交差検証の平均）")
    show(print, Experiments.evaluateSurvived(Paths.get(dataDir, "Survived.csv")), 4)
    print(s"cinema（線形回帰、${Experiments.NSplits} 分割交差検証の平均）")
    show(print, Experiments.evaluateCinema(Paths.get(dataDir, "cinema.csv")), 2)

  private def show(print: String => Unit, scores: Vector[(String, Double)], decimals: Int): Unit =
    scores.foreach((name, score) =>
      print(s"  $name: ${String.format(Locale.ROOT, s"%.${decimals}f", score)}")
    )
